src/semaphore.c

```diff
@@ -278,6 +278,31 @@ dispatch_semaphore_signal(dispatch_semaphore_t dsema)
 	return _dispatch_semaphore_signal_slow(dsema);
 }
 
+#ifdef USE_POSIX_SEM
+static dispatch_once_t sem_epoch_pred;
+static dispatch_time_t sem_epoch_diff;
```

Comment:
that should be `uint64_t`, `dispatch_time_t` is something else and cannot be used in arithmetics